src/api/SetTokenAPI.ts

```diff
@@ -152,6 +152,51 @@ export class SetTokenAPI {
     return quantity.mod(naturalUnit).eq(ZERO);
   }
 
+  /**
+   * Convenience function to calculate the units of components transferred given an issue quantity.
+   *
+   * @param  setAddress    Address of the Set
+   * @param  quantity      Quantity to be checked
+   * @return boolean       List of units
+   *
+   */
+  public async calculateComponentUnitsTransfered(setAddress: Address, quantity: BigNumber): Promise<BigNumber[]> {
+    const [naturalUnit, componentUnits] = await Promise.all([
+      this.setToken.naturalUnit(setAddress),
+      this.setToken.getUnits(setAddress),
+    ]);
+
+    return _.map(componentUnits, componentUnit => calculatePartialAmount(componentUnit, quantity, naturalUnit));
+  }
+
+  /**
+   * Convenience function to calculate the units of component transferred given an issue quantity.
+   *
+   * @param  setAddress        Address of the Set
+   * @param  componentAddress  Address of the component
+   * @param  quantity          Quantity to be checked
+   * @return boolean           Unit transfered in BigNumber format
+   *
+   */
+  public async calculateComponentUnitTransfered(
+    setAddress: Address,
+    componentAddress: Address,
+    quantity: BigNumber,
+  ): Promise<BigNumber> {
+    const [naturalUnit, componentUnits, components] = await Promise.all([
+      this.setToken.naturalUnit(setAddress),
+      this.setToken.getUnits(setAddress),
+      this.setToken.getComponents(setAddress),
+    ]);
+
+    const componentIndex = _.indexOf(components, componentAddress);
+    if (componentIndex < 0) {
+      throw new Error('Component `${componentAddress} is not a component of Set ${setAddress}');
```

Comment:
Isn't there an error for this? Why are we defining a generic error and a custom error message? Let's be consistent here and either add it to the list of other errors.

Also, put the assertion at the top. Okay with:
1. Doing the `getComponents` promise twice.
2. Having the `assertCalculateComponentUnitTransfered` method return the components so you don't have to do the promise again.